doc/user/scanners.xml

```diff
@@ -194,7 +195,7 @@ def kfile_scan(node, env, path, arg):
     <para>
 
       The scanner function must
-      accept the four specified arguments
+      accept three or four specified arguments
```

Comment:
This is a bit confusing.. It should accept 4 arguments, where the last one should have a default value as it may not be passed in all situations?

Reply:
Yes... I don't know why I kept that change, it was unintended. Many scanner functions in our tree do accept only three arguments, though, so it's kind of weird.  I don't like this logic very much:
```py
        if self.argument is not _null:                                         
            node_list = self.function(node, env, path, self.argument)          
        else:                                                                  
            node_list = self.function(node, env, path) 
```
why not just always pass the four parameters?  not proposing to change, I assume this has been around forever, just grumbling. I guess it avoids a check for `None` or something...

Reply:
btw the call to the path function does the same:
```py
        if self.argument is not _null:                                         
            return self.path_function(env, dir, target, source, self.argument)                                                         
        return self.path_function(env, dir, target, source)  
```